packages/lib-polkadot/index.ts

```diff
@@ -0,0 +1,81 @@
+import { DfnsApiClient, DfnsError } from '@dfns/sdk'
+import { PolkadotSigner } from './polkadot-signer'
+import { u8aConcat, u8aToHex } from '@polkadot/util'
+import { GenerateSignatureResponse } from '@dfns/sdk/types/wallets'
+
+export type DfnsWalletOptions = {
+  walletId: string
+  dfnsClient: DfnsApiClient
+}
+
+const assertSignResponseSuccessful = (response: GenerateSignatureResponse) => {
+  if (response.status === 'Failed') {
+    throw new DfnsError(-1, 'signing failed', response)
+  } else if (response.status !== 'Signed') {
+    throw new DfnsError(
+      -1,
+      'cannot complete signing synchronously because this wallet action requires policy approval',
+      response
+    )
+  } else if (!response.signature) {
+    throw new DfnsError(-1, 'signature missing', response)
+  }
+}
+
+export class DfnsWallet {
+  private readonly dfnsClient: DfnsApiClient
+  private readonly walletId: string
+  public readonly signer: PolkadotSigner
+  public static supportedNetworks: string[] = ['Polkadot', 'PolkadotWestend', 'Kusama']
+
+  // Flag to serialize EdDSA signatures properly
+  private ed25519PositionSignatures: Uint8Array = new Uint8Array([0])
+
+  private constructor(public address: string, options: DfnsWalletOptions) {
+    this.dfnsClient = options.dfnsClient
+    this.walletId = options.walletId
+    this.signer = new PolkadotSigner(this);
+  }
+
+  public static async init(options: DfnsWalletOptions) {
+    const { walletId, dfnsClient } = options
+    const res = await dfnsClient.wallets.getWallet({ walletId })
+
+    if (res.status !== 'Active') {
+      throw new DfnsError(-1, 'wallet not active', { walletId, status: res.status })
+    }
+
+    if (!this.supportedNetworks.includes(res.network as string)) {
+      throw new DfnsError(-1, 'wallet is not bound to a Polkadot compatible network', {
+        walletId,
+        network: res.network,
+      })
+    }
+
+    return new DfnsWallet(res.address!, options)
+  }
+
+  public async signRaw(data: string, address: string) {
+    if (this.address !== address) {
+      throw new DfnsError(-1, 'address does not match the wallet used to initialize DfnsWallet', {
+        expectedAddress: this.address,
+        givenAddress: address
+      })
+    }
+
+    const response = await this.dfnsClient.wallets.generateSignature({
+      walletId: this.walletId,
+      body: { kind: 'Transaction', transaction: data, },
+    })
+
+    assertSignResponseSuccessful(response)
+
+    return u8aToHex(
```

Comment:
Is there a reason to use this instead of Buffer.Concat?

From what I've seen, u8aFunction are useful when you deal with exotic `U8aLike`. Here you only have Buffer

(Maybe I am missing something)